tools/python/data-generator/flight_data_generator.py

```diff
@@ -0,0 +1,408 @@
+
+        :param start_date: The start date to use when generating records. This cannot be older in hours
+            than the memory retention period in hours value for the table.
+        :param end_date: The end date to use when generating records. The maximum end date
+            Timestream for LiveAnalytics allows is 15 minutes in the future.
+        :param reporting_frequency: The frequency that records are generated by all entities, for example,
+            every 2 seconds, every 5 hours, etc.
+        :param num_entities: The number of entities that will report for each timestamp, for example,
+            the number of servers or number of stocks.
+        :param precision: The precision to use for record timestamps. Valid options are "MILLISECONDS",
+            "SECONDS", and "MICROSECONDS".
+        :param generate_unique_options_fallback: Whether to generate random strings for dimension values
+            after all values in a dimension template's "unique_options" array have been used.
+        """
+        num_flight_ids = len(self.dimension_templates[0]["unique_options"])
+        if num_entities > num_flight_ids and not generate_unique_options_fallback:
+            raise Exception(f"num_entities ({num_entities}) was greater than the number of flight IDs ({num_flight_ids})")
+
+        # 1. Construct entities with departure & destination airports
+        self.entities = []
+        for _ in range(num_entities):
+            entity = {"latest_measures": {}}
+            for dimension_template in self.dimension_templates:
+                dimension_value_length = 10
+                if "value_length" in dimension_template:
+                    dimension_value_length = dimension_template["value_length"]
+                if "unique_options" in dimension_template:
+                    if len(dimension_template["unique_options"]) > 0:
+                        dimension_value = dimension_template["unique_options"][0]
+                        dimension_template["unique_options"].pop(0)
+                    elif generate_unique_options_fallback:
+                        dimension_value = self._generate_random_string(dimension_value_length)
+                elif "random_options" in dimension_template:
+                    dimension_value = random.choice(dimension_template["random_options"])
+
+                elif "unique_options" not in dimension_template and "random_options" not in dimension_template:
+                    dimension_value = self._generate_random_string(dimension_value_length)
+
+                entity[dimension_template["name"]] = dimension_value
+
+            # Pick departure and destination airports
+            departure_index = random.randint(0, len(self.airports) - 1)
+            destination_index = random.randint(0, len(self.airports) - 1)
+            while destination_index == departure_index:
+                destination_index = random.randint(0, len(self.airports) - 1)
```

Comment:
```suggestion
            destination_index = random.choice(list(set(range(len(self.airports))) - {departure_index}))
```

Even though the probability of picking the same departure and destination index is low, using sets eliminates the possibility of picking the same indices.

Reply:
The current approach also guarantees that we wont pick the same index (via rejection sampling), and is a bit more readable IMO. Let me know if you still prefer using sets here

Reply:
What I meant by "picking the same departure and destination index" is the case where the while loop is entered. Although less readable, I prefer using sets in this situation because it avoids unnecessary looping and avoids an infinite loop if `self.airports` contains one airport.